addons/mail/static/src/core/common/thread_model.js

```diff
@@ -1084,6 +1084,7 @@ export class Thread extends Record {
         if (pushState) {
             router.pushState({ active_id: activeId });
         }
+        this.store.env.services.action?.currentController?.config?.setDisplayName(this.displayName);
```

Comment:
This appears to not be called properly when using the browser back button when going back to a channel coming from a form view for example.

Reply:
@seb-odoo I did not find the exact reason why the name is not saved when using browser buttons. I checked in `web/action_service.js` and `web/control_panel.js` but it may require a deeper investigation. Anyway, I solved the issue using the same ["trick" as the `FormController` component](https://github.com/odoo/odoo/blob/8d915c84bb269ada9aee38681f049d3d8cf88e42/addons/web/static/src/views/form/form_controller.js#L260-L262)